src/main/java/org/rumbledb/runtime/operational/TreatIterator.java

```diff
@@ -45,8 +45,10 @@ public TreatIterator(
         this.iterator = iterator;
         this.sequenceType = sequenceType;
         this.shouldThrowTreatException = shouldThrowTreatException;
-        this.itemType = this.sequenceType.getItemType();
-        this.sequenceTypeName = ItemTypes.getItemTypeName(this.itemType.getType().toString());
+        if (!this.sequenceType.isEmptySequence()) {
+            this.itemType = this.sequenceType.getItemType();
```

Comment:
I see a null pointer exception risk here. this.itemType is not initialized (is null) when sequenceType is emptySequence.

At line 116, this.nextResult.isTypeOf(this.itemType) call is made.

For example, if this.nextResult is a BooleanItem, isTypeOf method on the boolean item does not handle null parameters and it will cause a NPE. 

It may be that execution does not reach there if sequence is specified to be an emptySequence. But it would take me a bit more time to fully grasp all the implications, so I wanted to get your opinion as well.

Reply:
Yes, thank you for being thorough. I actually had this in mind and did a few more fixes yesterday, but github had internal server errors. My last commit fixes issues related to this.

Reply:
This is discussed offline. An error check is performed in ther code preventing the execution from reaching this NPE.